lib/Config.js

```diff
@@ -280,6 +280,46 @@ function requestsConfigAssert(requestsConfig) {
     }
 }
 
+function bucketNotifAssert(bucketNotifConfig) {
+    assert(Array.isArray(bucketNotifConfig),
+        'bad config: bucket notification configuration must be an array');
+    bucketNotifConfig.forEach(c => {
+        const { resource, type, host, port, auth } = c;
+        assert(typeof resource === 'string',
+            'bad config: bucket notification configuration resource must be a string');
+        assert(typeof type === 'string',
+            'bad config: bucket notification configuration type must be a string');
+        assert(typeof host === 'string' && host !== '',
+            'bad config: hostname must be a non-empty string');
+        assert(Number.isInteger(port) && port > 0,
+            'bad config: port must be a positive integer');
+        assert(typeof auth === 'object',
+            'bad config: bucket notification auth must be an object');
+        const { cert, user, password } = auth;
+        if ((cert && user && password) || (!cert && !user && !password)) {
```

Comment:
It probably doesn't happen often but I think the first set should be `cert && (user || password)` as it would be weird to specify a cert and a user even if you forgot the password and vice versa.